src/Factory/ProductListViewFactory.php

```diff
@@ -21,13 +21,54 @@
 
 final class ProductListViewFactory implements ProductListViewFactoryInterface
 {
+    /** @var string */
+    private $productListViewClass;
```

Comment:
Won't it be better to use `Sylius\Component\Resource\Factory\Factory`? At least we can do it in another iteration together with some little refactoring of this code.